### ISSUE TITLE
[Grid - Sorting] Changes of `aria-sort` are not announced by screen readers

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3497]
### 😯 Current Behavior

This bug is related to NVDA, but fix for it may also improve a11y of sorting for other screen readers. 

Right now when NVDA user changes sorting via button - only changed button's name is being announced. User is not getting information about changed `aria-sort` value. Name of button that is being announced only inform about NEXT action of a button, so when sorting is set to 'ascend' then name of button is "Sort descending". This may be confusing to users when they only get information from the button without `aria-sort` from `<th>`.

https://github.com/user-attachments/assets/9a65f2ce-d35b-494e-85be-03ebd54c94a2

At the same time VoiceOver doesn't announce even the changed name of a button, so when changing sorting no feedback is announced to user automatically. User has to move focus to different element and then go back to button or `<tr>` to get information about next/current sorting.

https://github.com/user-attachments/assets/0751fc17-e535-41b0-85a8-9cd152d6b05b

### 🤔 Expected Behavior/possible solution?

It's recommended to add additional information about current `aria-sort` value that will be announced automatically after changing sorting. 

It can be done either by: 
1. Adding live region using `sl-announcer` that would announce 'COLUMN NAME sorting ascending/descending/removed'. This announcement should be read after each change done by user.
2. Adding more information to `aria-label` of button used to change sorting. Label should end with '[...] Now sorting ascending/descending/removed'.

Option 1 is preferred as it'll make VoiceOver to say more helpful information without need to change focus forth and back.

Examples where `aria-sort` is being announced after it change: 
1. https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/
2. https://react.carbondesignsystem.com/?path=/story/components-datatable-sorting--default

Examples are not as accessible as they can be, because all of them do not announce changed label of button on VoiceOver and that's why I recommend to add `sl-announcer` to fix that issue.

### 🖥️ Steps to Reproduce

1. Go to https://storybook.sanomalearning.design/?path=/story/grid-sorting--basic
2. With turned on screen reader place focus on first table header with sorting button
3. Using `Space` change sorting 
4. Check if current `aria-sort` value is announced automatically when it's changed.

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/grid-sorting--basic

### ⚖️ Impact

Not a huge problem, we have a workaround

### Framework

Other

### What browsers are you seeing the problem on?

All

### Component version

0.7.4

### 👤 Your name

Kacper on behalf of Katarzyna Werner-Mozolewska

### 🧢 Your product/team

Learning Analitics
